### PR TITLE
fix clear events when nothing is scheduled

### PR DESCRIPTION
**Pull request by Fr0stbyteR**
@owengc 
When there's nothing in the _pendingEvents, the Promise doesn't resolve. Trying to fix.

### CODE REVIEW

**Comment by owengc** [comment]
nice catch @Fr0stbyteR ! looks good to me 👍

**Review by owengc — APPROVED** [review]

File: src/WamNode.js [hunk 1]
```diff
@@ -248,6 +248,8 @@ export default class WamNode extends AudioWorkletNode {
 			if (ids.length) {
 				this._pendingResponses[id] = resolve;
 				this.port.postMessage({ id, request });
+			} else {
```
Comment (owengc) [inline src/WamNode.js]: i’m not sure what’s preferable from a ‘best practices’ standpoint, but another option would be to do something like this right?
```
async clearEvents() {
  const request = 'remove/events';
  const id = this._generateMessageId();
  const ids = Object.keys(this._pendingEvents);
  if (ids.length) {
    return new Promise((resolve) => {
      this._pendingResponses[id] = resolve;
      this.port.postMessage({ id, request });
    }).then((clearedIds) => {
      clearedIds.forEach((clearedId) => {
        this._pendingEvents[clearedId]();
        delete this._pendingEvents[clearedId];
      });
    });
  }
}
```
seems a little cleaner to me

**Review by owengc — APPROVED** [review]
i'm good with your solution here, but I left a comment with an alternative implementation of your fix which seems a little cleaner to me

File: src/WamNode.js (hunk 1 above)
Comment (Fr0stbyteR) [reply src/WamNode.js]: changed to this one, better than my quick fix ;)

**Pull request merged**